config/routes.rb

```diff
@@ -1,4 +1,35 @@
 Rails.application.routes.draw do
+
+  root "welcome#index"
+
+  get 'welcome/index'
+
+  resources :products, :only => [:index, :show] do
+    resources :order_items, :only => [:create, :update, :destroy]
+    resources :reviews
+    post 'add_to_cart', on: :member
+  end
+  get 'search' => 'products#search', as: 'search_product'
+
+  resources :users, :only => [:new, :create, :show, :edit, :update] do
+    resources :reviews
+    scope module: :users do
+      resources :products
+      resources :fulfillment, :only => [:index, :show]
+    end
+  end
+
+  resources :orders, :only => [:show, :edit, :update]
+
+  resources :order_items, :only => [:show]
+
+  resources :sessions, :only => [:create]
+  delete "/logout" => "sessions#destroy"
+  get    "/login"  => "sessions#new"
+
+  resources :categories, :only => [:new, :create]
+
+
```

Comment:
I like how much care was put into restricting the routes here.
It is always good to do a double check and make sure that you don't have any unhandled routes.

Typing `rake routes` in the terminal and then checking one by one is a good way to make sure that you have the right method for each one or remind you to limit it.